### ISSUE TITLE
Multi-project builds slower from command-line than from sbt console

### ISSUE THREAD

**Issue by varju** [sbt/sbt#838]
I've been working on converting a monolithic project up into multiple sub-projects, and noticed that building from the command-line was notably slower than building the same thing from within the sbt console.

From the command line, running `sbt clean test:compile clean test:compile` (building twice to ensure caches are warm); second compile completes in 498s.  From the sbt console, running `;clean;test:compile;clean;test:compile`; second compile completes in 52s.

This issue only appears to affect multi-project builds.  Doing the same test with a standalone project shows identical compile times from the command-line and the sbt console.

This issue was reproduced using sbt 0.12.4.

**Comment by harrah** [comment]
I tried this with akka and cannot reproduce.  The second compile is around 260s in both cases.

There is almost no distinction between running commands from the command line or interactively.  In one case, the next command is read from stdin via JLine, but that is it.  (The default number of frames printed for a stack trace is different, but there are presumably no traces being printed for a successful compilation.)

**Comment by varju** [comment]
Weird.  Must be something special about my codebase.  I'll see if I can come up with an example I can share that demonstrates this.

**Comment by varju** [comment]
Although not quite to the same extreme, I just reproduced the problem with the Play! source code.  1073s for the console build and 1510s from the command-line.  I did this twice just to be sure it wasn't caused by some unrelated background process during the slower build.

Just to be sure there wasn't something weird about my machine, I tried building Akka like you did and did not see the problem there.

My project is based on Play, and extends play.Project; is it plausible that something pulled in by their project configuration is causing this?
